relay-server/src/actors/events.rs

```diff
@@ -1696,11 +1700,25 @@ impl Handler<HandleEnvelope> for EventManager {
                     outcome_producer.do_send(TrackOutcome {
                         timestamp: Instant::now(),
                         scoping: *scoping.borrow(),
-                        outcome,
+                        outcome: outcome.clone(),
                         event_id,
                         remote_addr,
                         category,
-                    })
+                        quantity: 1,
+                    });
+
+                    let envelope_summary = envelope_summary.borrow();
+                    if envelope_summary.attachment_quantity > 0 {
```

Comment:
would it ever be possible to have an attachment with quantity 0?

Reply:
Strictly speaking, it is possible to have empty attachments. `EnvelopeSummary` assumes length 1 in this case to ensure that we invoke rate limiting code, and now also create an outcome:

https://github.com/getsentry/relay/blob/6a42cb4c76203ff63738e161a4d73481ef67dd82/relay-server/src/utils/rate_limits.rs#L148

We do not count meta data, such as the file name.